evm/src/main/scala/pravda/evm/translate/opcode/JumpDestinationPrepare.scala

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2018  Expload.com
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package pravda.evm.translate.opcode
+
+import pravda.evm.EVM
+import pravda.evm.EVM.JumpDest
+import pravda.vm.asm.Operation
+import pravda.vm.{Opcodes, asm}
+import pravda.vm.asm.Operation.PushRef
+
+object JumpDestinationPrepare {
+
+  def jumpDestToOps(op: (EVM.JumpDest, Int)): List[asm.Operation] =
+    op match {
+      case (JumpDest(addr), ind) =>
+        List(
+          asm.Operation.Label(getNameByNumber(ind)),
+          asm.Operation(Opcodes.DUP),
+          pushBigInt(addr),
+          asm.Operation(Opcodes.EQ),
+          asm.Operation(Opcodes.NOT),
+          Operation.JumpI(Some(getNameByNumber(ind + 1))),
```

Comment:
`ind + 1`? Are you sure it's correct?

Reply:
You should write a test for jumps

Reply:
[Here](https://github.com/expload/pravda/pull/238/files#diff-8d84434a1d81d866cf7389c884b00c16R166)

Reply:
[Here](https://github.com/expload/pravda/pull/238/files#diff-8d84434a1d81d866cf7389c884b00c16R166)

Reply:
[Here](https://github.com/expload/pravda/pull/238/files#diff-8d84434a1d81d866cf7389c884b00c16R166)

Reply:
[here](https://github.com/expload/pravda/pull/238/files#diff-8d84434a1d81d866cf7389c884b00c16R166)

Reply:
[here](https://github.com/expload/pravda/pull/238/files#diff-8d84434a1d81d866cf7389c884b00c16R166)

Reply:
Your test doesn't check case with several `Jump`s, so it doesn't `ind + 1` behavior actually.